client/support/support-user/localforage-bypass.js

```diff
@@ -0,0 +1,242 @@
+/**
+ * External dependencies
+ */
+import debugModule from 'debug';
+
+const debug = debugModule( 'calypso:support-user' );
+
+// This module defines a custom localForage driver which bypasses all persistent
+// storage. Any calls to read/write data using localForage instead access a temporary
+// in-memory store which is lost on page reload. This driver is used to sandbox
+// a user's data while support-user is active, ensuring it does not contaminate the
+// original user/vice versa.
+// Copied with modifications from localForage source: localforage/test/dummyStorageDriver.js
+
+var dummyStorage = {};
+
+// Config the localStorage backend, using options set in the config.
+function _initStorage( options ) {
+	const self = this;
+
+	let dbInfo = {};
+	if ( options ) {
+		for ( let i in options ) {
+			dbInfo[ i ] = options[ i ];
+		}
+	}
+
+	dummyStorage[ dbInfo.name ] = dbInfo.db = {};
+
+	self._dbInfo = dbInfo;
+	return Promise.resolve();
+}
+
+function clear( callback ) {
+	debug( 'localForage bypass', 'clear' );
+
+	const self = this;
+	const promise = new Promise( function( resolve, reject ) {
```

Comment:
Two other options would be to use `.bind` or fat arrow, to avoid creating the self const:
```javascript
const promise = new Promise( function( resolve, reject ) => {
//...
} ).bind( this ) );
```
```javascript
const promise = new Promise( ( resolve, reject )  => {
```